addons/payment_xendit/tests/test_payment_transaction.py

```diff
@@ -12,6 +12,30 @@
 @tagged('post_install', '-at_install')
 class TestPaymentTransaction(PaymentHttpCommon, XenditCommon):
 
+    def test_no_item_missing_from_rendering_values(self):
+        """ Test that the when redirect flow is triggered, rendering_values contain the
+        API_URL corresponding to the response of API request. """
+        tx = self._create_transaction(flow='redirect')
+        with patch(
+            'odoo.addons.payment_xendit.models.payment_provider.PaymentProvider'
+            '._xendit_make_request', return_value={'invoice_url': 'https://dummy.com'}
+        ):
+            rendering_values = tx._get_specific_rendering_values(None)
+        self.assertDictEqual(rendering_values, {'api_url': 'https://dummy.com'})
+
+    def test_empty_rendering_values_if_direct(self):
+        """ Test that if it's a card payment (like in direct flow), rendering_values should be empty
+        and no API call should be committed in the process"""
+        card_pm = self.env.ref('payment.payment_method_card').id
+        tx = self._create_transaction(flow='direct', payment_method_id=card_pm)
+        with patch(
+            'odoo.addons.payment_xendit.models.payment_provider.PaymentProvider'
+            '._xendit_make_request', return_value={'data': {'link': 'https://dummy.com'}}
+        ) as mock:
+            rendering_values = tx._get_specific_rendering_values(None)
+            mock.assert_not_called()
+        self.assertDictEqual(rendering_values, {})
+
```

Comment:
Could you please move the `test_no_input_missing_from_redirect_form` test here, so that it's close to the other related tests?

Reply:
The `test_no_input_missing_from_redirect_form` was added preivously https://github.com/odoo-dev/odoo/blob/d4104c3cb92b2addb96de9e5018a054d8a86d293/addons/payment_xendit/tests/test_payment_transaction.py#L67 
(I didn't remember adding this before but I think you were the one who added it before :rofl: )

Reply:
Sure, but there were only two tests at the time: one for the redirect values and one for the invoice request. With the addition of two redirect values tests, the file is structured like this: redirect, redirect, invoice, redirect. I think it'd be easier to read if it was: redirect, redirect, redirect, request.

Reply:
Noted, I will rearrange the file